Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/opensaml/HOKSAMLAssertionBuilder.java

```diff
@@ -288,28 +287,28 @@ protected Conditions createConditions(final CallbackProperties properties) {
 
         // Not a *single* one of these methods ever returns a list with more than one element.
         // so in that case we can eventually drop the whole "List" thing going on and just add individual elements.
-        statements.addAll(createAuthenicationStatements(properties));
+        statements.add(createAuthenicationStatements(properties));
 
         // The following 6 statements are required for NHIN Spec
-        statements.addAll(createSubjectIdAttributeStatement(properties));
-        statements.addAll(createOrganizationAttributeStatements(properties));
-        statements.addAll(createSubjectRoleStatement(properties));
-        statements.addAll(createPurposeOfUseStatements(properties));
-        statements.addAll(createHomeCommunityIdAttributeStatements(properties));
-        statements.addAll(createOrganizationIdAttributeStatements(properties));
+        statements.add(createSubjectIdAttributeStatement(properties));
+        statements.add(createOrganizationAttributeStatements(properties));
+        statements.add(createSubjectRoleStatement(properties));
+        statements.add(createPurposeOfUseStatements(properties));
+        statements.add(createHomeCommunityIdAttributeStatement(properties));
+        statements.add(createOrganizationIdAttributeStatement(properties));
 
         // These are optional and may be omitted in the NHIN Spec
-        statements.addAll(createResourceIdAttributeStatements(properties));
-        statements.addAll(createNPIAttributeStatements(properties));
+        statements.add(createResourceIdAttributeStatement(properties));
```

Comment:
Since these methods for optional fields can now return a null value here instead of an empty list, we should have an appropriate null check before adding it to the list.